conjure-runtime/src/request.rs

```diff
@@ -117,6 +118,8 @@ pub(crate) struct Request<'a> {
     pub(crate) headers: HeaderMap,
     pub(crate) body: Option<Pin<Box<ResetTrackingBody<dyn Body + Sync + Send + 'a>>>>,
     pub(crate) idempotent: bool,
+    pub(crate) propagate_qos_errors: bool,
```

Comment:
nit: in the future, how do you feel about using enums to control behavior instead of booleans? i.e. `OnQosError.PROPAGATE` vs `OnQosError.RETRY`?

Reply:
Yeah, the next thing I'm planning on working on is switching to something more like Dialogue's ClientFactory with its enums.